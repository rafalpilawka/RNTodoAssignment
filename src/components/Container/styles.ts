import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  wrapper: {
    height: '100%',
    backgroundColor: 'transparent',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
