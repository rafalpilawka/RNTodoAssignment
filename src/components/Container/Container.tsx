import React, {ReactElement, ReactNode} from 'react';
import {SafeAreaView, ViewStyle} from 'react-native';
import {styles} from './styles';

interface Props {
  children?: ReactNode;
  style?: ViewStyle;
}

export const Container = ({children, style}: Props): ReactElement => {
  return (
    <SafeAreaView style={[styles.wrapper, style]}>{children}</SafeAreaView>
  );
};
