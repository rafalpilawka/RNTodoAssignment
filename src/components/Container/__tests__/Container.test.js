import React from 'react';
import {shallow} from 'enzyme';
import {Container} from 'src/components/Container/Container';
import renderer from 'react-test-renderer';
import {Text} from 'react-native';

describe('Container', () => {
  let component;
  const text = 'Test';

  beforeEach(() => {
    component = shallow(
      <Container>
        <Text>{text}</Text>
      </Container>,
    );
    jest.clearAllMocks();
  });
  it('should match to snapshot', () => {
    const component = renderer.create(<Container />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render correct children', () => {
    const labelElement = component.find({children: text});
    expect(labelElement).toHaveLength(1);
  });
  it('not render empty children array', () => {
    const componentDive = shallow(<Container />).dive();
    const safeAreaChildren = componentDive.props().children;
    expect(safeAreaChildren).toBeUndefined();
  });
});
