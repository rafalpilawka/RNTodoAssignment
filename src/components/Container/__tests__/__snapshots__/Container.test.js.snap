// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match to snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;
